### ISSUE TITLE
Step 2: Parse First Name - Final Project

### ISSUE THREAD

**Issue by Dimetry-Adel** [Watts-College/cpp-527-fall-2021#67]
Hi,

Also, when running the functions for step 2.
```
top.25 <- head( d$Full.Name,25 )
first.25 <- get_first_name( name=top.25 )

data.frame( top.25, first.25 ) %>% pander()
```

This error appears.
![image](https://user-images.githubusercontent.com/76553425/135770907-d8500782-c47a-473c-aea1-e1cec60bae61.png)

Please, could you have an explanation for it?
Thanks

**Comment by lecy** [comment]
You need to write the name parsing function called **get_first_name()** that will return only the first name. 

The code is showing you what it should look like when working properly (full name vs first name): 

![image](https://user-images.githubusercontent.com/1209099/135775934-41852764-7689-4cc7-87aa-64305face7b4.png)

**Comment by voznyuky** [comment]
Are we using split strings in this or am I way off?

**Comment by lecy** [comment]
Yep, you can build from what you did in lab 3 finding the first or last words in titles.

**Comment by voznyuky** [comment]
Lab 3 is the one I struggled on the most. This is what I have so far, although it is just breaking up the strings. How do I pull the 2nd word from each string, when there are multiple strings? I might need a little extra help on this step of the assignment.

get_first_name <- function( x )
{
  first.names <- strsplit( d$Full.Name, " " )
  return( first.names )
}

get_first_name()

**Comment by lecy** [comment]
What is the right delimiter to use for the split? Is it a space in this context?

**Comment by lecy** [comment]
Describe your pseudo code

**Comment by lecy** [comment]
Here is some code to get you started: 

https://github.com/Watts-College/cpp-527-fall-2021/issues/83#issuecomment-938342151

For the second part you basically repeat the same steps, but use a space as the split value then grab the first element in each vector instead of the second. This will drop middle names for cases like Jose: 

```
"ARQUIZA, Jose Maria Reynaldo Apollo"
```

**Comment by lecy** [comment]
If writing code that doesn't work makes you humble then at this point I might be a saint :-)
